sw/device/lib/crypto/drivers/aes_test.c

```diff
@@ -0,0 +1,107 @@
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+static const char kSecretKey[] = "Secret test key.";
+
+static const char kPlaintext[] =
+    "Muchos años después, frente al pelotón de fusilamiento, el coronel "
+    "Aureliano Buendía había de recordar aquella tarde remota en que su "
+    "padre lo llevó a conocer el hielo. Macondo era entonces una aldea de "
+    "veinte casas de barro y cañabrava construida a la orilla de un río de "
+    "aguas diáfanas que se precipitaban por un lecho de piedras pulidas, "
+    "blancas y enormes como huevos prehistóricos. El mundo era tan reciente, "
+    "que muchas cosas carecían de nombre, y para mencionarlas había que "
+    "señalarlas con el dedo.";
+
+// AES-CTR encrypted using the key as the IV.
+static const uint32_t kCiphertext[] = {
+    0x9663bf38, 0x536926f4, 0x0e2f0ce2, 0x59e4475d, 0x468e0218, 0xcca04587,
+    0x7604901b, 0xf1fd87fd, 0xa814254e, 0x83c9dd62, 0xaad7e8eb, 0xdbb7528b,
+    0xbadc51e1, 0x5128fc8c, 0x0df70958, 0x958d8fe6, 0x9c6658e0, 0xdb8f150a,
+    0x55d3821d, 0xf0fa39d2, 0x3c827fba, 0x410b5f1e, 0x6f183ad2, 0x4f7b0eb4,
+    0xb9aa403e, 0x51c531b8, 0x695a344e, 0x3e749fdf, 0xc96a3dad, 0xe1b96919,
+    0x8381e169, 0x66101610, 0x0a1ccf3b, 0xa640bce2, 0xf138225f, 0xf1660827,
+    0x88e2f3a7, 0x4d23a5b1, 0xe45708e5, 0x67f2d301, 0x66a48d4d, 0xacbbf79a,
+    0x750dd0b3, 0x62931566, 0x7ed5bbad, 0x83d59c1e, 0x41755aa1, 0x18955d3f,
+    0xbb2a1fe1, 0x908c02e1, 0x108aa3b7, 0x9732b3ce, 0x8383d89f, 0xdea7b27c,
+    0x92dee579, 0xf50060f0, 0xef754f42, 0xec5b9b13, 0xb5dbe7fc, 0x21119fe5,
+    0x5da3e9bd, 0xa971de87, 0x530ba4d2, 0x73b62820, 0xb65839d4, 0xe6720fa2,
+    0xde95b28c, 0xb8020db1, 0xeb4edb9e, 0x2d24a18e, 0xacb7ca97, 0x326fce63,
+    0xda90c391, 0xb5ead3e7, 0xf60879a4, 0xf96d1bdc, 0x0f291e0d, 0xc2178eea,
+    0x45bf2b76, 0x571b4859, 0xcce364fb, 0x8909a2ab, 0x225ef2bb, 0xc23e7b84,
+    0xdcfa00d7, 0xc3d475ae, 0xc007c9ef, 0xe29ff430, 0xbae4d34d, 0xceed477d,
+    0xc784b08d, 0x53928311, 0x42ea5414, 0xa6f098dd, 0x9fe5c2ab, 0x7bc2a2da,
+    0x5a57d069, 0xefc5e4d1, 0x7d94ffee, 0xd07a2437, 0xfa0ff98d, 0x817cbfbd,
+    0x34acb057, 0x11e15386, 0x30daae97, 0x818fa7de, 0x4f678086, 0x2b1f50fe,
+    0x412007f7, 0x9e19b989, 0xb246848d, 0x1be66862, 0x55611aa9, 0x3da44d0c,
+    0x4d67a944, 0x7fcf38e7, 0xf411193c, 0x01a49b20, 0x4d8a230d, 0x63bee9cc,
+    0x3938f063, 0x4565f60d, 0xce9286b8, 0xaeaa93b6, 0x4d43e052, 0x2d0e6cf7,
+    0x998d1fd8, 0x55e8bc50, 0x73d19e17,
+};
+
+const test_config_t kTestConfig;
+bool test_main(void) {
+  uint32_t iv[4];
+  memcpy(iv, kSecretKey, ARRAYSIZE(kSecretKey));
+
+  uint32_t share0[4];
+  uint32_t share1[4];
+  for (int i = 0; i < ARRAYSIZE(share0); ++i) {
+    if (i % 2 == 0) {
+      share0[i] = ~iv[i];
+      share1[i] = -1;
+    } else {
+      share1[i] = ~iv[i];
+      share0[i] = -1;
+    }
+  }
```

Comment:
You may know it but others don't. They will look at this code and think this is the way to do it because the OpenTitan guys for sure know what they're doing. The very least you should do here is describing what you do and why, AND clearly specifying that this is not how it should be done in a real application because it's not secure.